apps/judicial-system/web/src/routes/Prosecutor/Indictments/Overview/Overview.tsx

```diff
@@ -239,6 +255,21 @@ const Overview: React.FC<React.PropsWithChildren<unknown>> = () => {
           nextIsDisabled={
             userCanSendCaseToCourt && !indictmentConfirmationDecision
           }
+          hideActionButton={
+            workingCase.state === CaseState.WAITING_FOR_CANCELLATION
+          }
+          actionButtonText={formatMessage(strings.askForCancellationButtonText)}
+          actionButtonColorScheme={'destructive'}
+          actionButtonIsDisabled={
+            !(
```

Comment:
This code seems a bit hard to read. Consider extracting this to a function instead, something like

```
const isActionButtonDisabled = () => {
    const isSubmittedOrReceived = [CaseState.SUBMITTED, CaseState.RECEIVED].includes(workingCase.state)
    const hasIndictmentDecision = workingCase.indictmentDecision
    const hasState = workingCase.state
    
    return isSubmittedOrReceived && hasIndictmentDecision && !hasState 
}
```

Check the actual implementation though 😅

Reply:
I used a different approach and introduced a variable with a descriptive name.